src/type-extensions/function.ts

```diff
@@ -70,7 +71,7 @@ class FunctionTypeExtension extends TypeExtension<FunctionType, FunctionDecorati
 
     public pushValue(thread: Thread, decoratedValue: Decoration<FunctionType, FunctionDecoration>): boolean {
         const { target, options } = decoratedValue
-        if (typeof target !== 'function') {
+        if (typeof target !== 'function' && !(options as UserDataDecorationOptions).reference) {
```

Comment:
This is odd, if reference is set then userdata is a higher priority than this so this pushValue shouldn't be called, at least in theory

Reply:
You are right, I forgot about the priority logic :O